state-chain/pallets/cf-environment/src/lib.rs

```diff
@@ -688,7 +688,13 @@ impl<T: Config> Pallet<T> {
 	}
 
 	pub fn get_sol_nonce_and_account() -> Option<DurableNonceAndAccount> {
-		SolanaAvailableNonceAccounts::<T>::mutate(|nonce_and_accounts| nonce_and_accounts.pop())
+		let nonce_and_account = SolanaAvailableNonceAccounts::<T>::mutate(|nonce_and_accounts| {
+			nonce_and_accounts.pop()
+		});
+		nonce_and_account.map(|(account, nonce)| {
+			SolanaUnAvailableNonceAccounts::<T>::insert(account, nonce);
```

Comment:
Do we need to store the nonce value here? It looks like it's never used - we could just store a map of `<Account, ()>`?

Reply:
The reason we need to store the used nonce here is because of how current nonce witnessing works. the cfe checks if the witnessed nonce is the same as the current in-use nonce.